tests/drivers/uart/uart_async_api/src/test_uart.h

```diff
@@ -27,4 +27,9 @@
 #define UART_NODE DT_CHOSEN(zephyr_console)
 #endif
 
+#if DT_NODE_HAS_COMPAT(UART_NODE, zephyr_uart_emul)
+#include <zephyr/drivers/serial/uart_emul.h>
+#define DUT_IS_UART_EMUL 1
+#endif
```

Comment:
This doesn't seem right. It is possible that there is emulated UART, but the test is actually running with a real UART on hardware.

Reply:
Thats why i checked if `UART_NODE`, which is the one used in the testcases, is the emulated uart.

Reply:
This should not be needed, as the emulated device driver should behave like any other hardware device, if the test suite must be changed to support the emulated uart, it is not emulating a uart device properly :)

Reply:
I removed this. The special case for the abort test is still there, though